image-assets/upload-image-to-aws.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+
+# Given a raw image created with Auroraboot, this script will upload it to the speficied AWS account as a public AMI.
+# Docs:
+# https://docs.aws.amazon.com/vm-import/latest/userguide/required-permissions.html
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/creating-an-ami-ebs.html#creating-launching-ami-from-snapshot
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-ami-boot-mode.html
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/launch-instance-boot-mode.html
+
+set -e
+set -o pipefail
+
+checkArguments() {
+  if [ $# -lt 1 ]; then
+    echo "Error: You need to specify the cloud image to upload."
+    echo "Usage: $0 <cloud-image>"
+    exit 1
+  fi
+
+  local file="$1"
+
+  if [ ! -f "$file" ]; then
+    echo "Error: File '$file' does not exist."
+    exit 1
+  fi
+
+  if ! file "$file" | grep -q 'DOS/MBR boot sector'; then
+    echo "Error: File '$file' is not a raw image."
+    exit 1
+  fi
+}
+
+checkEnvVars() {
+  if [ -z "$AWS_PROFILE" ] || [ -z "$AWS_REGION" ] || [ -z "$AWS_S3_BUCKET" ]; then
+    echo "Error: AWS_PROFILE, AWS_REGION and AWS_S3_BUCKET environment variables must be set."
+    exit 1
+  fi
+}
+
+AWS() {
+  aws --profile $AWS_PROFILE --region $AWS_REGION "$@"
```

Comment:
do we need the aws binary in the image to use it, right?

Reply:
yes but we discussed this with Itxaka and decided we should move this script to the kairos repo (only to be used in CI). We don't want to support this script officially so we shouldn't ship it with Auroraboot.